source/mainnet/smart-contracts/tutorials/sponsoredTransactions/sponsoredTransactionsSmartContract.rst

```diff
@@ -0,0 +1,247 @@
+    also known as public-key cryptography, where you generate a pair of related keys --
+    one public key and one private key. The sponsoree encrypts/signs a message with its private key
+    and anyone can decrypt the sponsoree's message using its public key. The public key is shared with other entities (e.g., the smart contract) while the private key is only known to the sponsoree.
+    The smart contract needs to know the public key to be able to verify that the sponsoree signed the message with the corresponding private key.
+
+.. note::
+
+    The |bw| prepends the message with the account address (``signer``) and 8 zero bytes before signing it.
+    This ensures that signed messages can be distinguished from transactions that are equally signed by the account address in the |bw| because transactions don't start with an account address.
+    The smart contract logic reverse engineers the |bw| implementation by equally prepending the account address (signer) and 8 zero bytes to the message.
+
+.. note::
+
+   Concordium accounts can be multi-sig and each account has at least one ``public key`` and at least
+   one ``private key`` associated with it.
+   For this sponsored transactions example, the accounts in the |bw| (or |mw-gen2|)
+   have exactly one ``public key`` and exactly one ``private key`` (no multi-sig accounts).
+   You use your ``private key`` to sign the message in the wallet and your ``public key`` is
+   used in the smart contract to verify that this signature was generated in the wallet with the associated ``private key``.
+   Never share your ``private key``.
+
+``supportsPermit`` function
+===========================
+
+This function can be queried with a list of given entrypoint names.
+The response contains corresponding results for each entrypoint,
+where the result is either “Entrypoint is not supported and can not be invoked via
+the permit function using the sponsored transaction mechanism” or “Entrypoint is supported and
+can be invoked via the permit function using the sponsored transaction mechanism”.
+
+You can explore the function by invoking it with the ``concordium-client`` as follows:
+
+.. code-block:: console
+
+    $concordium-client contract invoke 4329 --entrypoint supportsPermit --parameter-json supportsPermit.json --grpc-port 10000 --grpc-ip node.testnet.concordium.com
+
+For example, this ``supportsPermit.json`` file results in the below screenshot.
+
+.. code-block:: json
+
+    {"queries":["aaaa","transfer","updateOperator"]}
+
+.. image:: ./images/supportsPermit.png
+   :alt: Output of supportsPermit function
+   :align: center
+
+.. note::
+
+   Comprehensive instructions on how to download and set up ``concordium-client`` can be found in :ref:`Setup the development environment<setup-env>`.
+
+``registerPublicKeys`` function
+===============================
+
+Concordium smart contracts currently have no way to query the corresponding
```

Comment:
Could we make a record of this sentence, maybe as an issue, so we know to revisit it when we add this feature?

Reply:
https://github.com/Concordium/concordium-rust-smart-contracts/issues/258